renderer.php

```diff
@@ -240,34 +240,38 @@ public function createform($data) {
             $record = $value->readsfrom;
             $tmpparam = str_replace(' ', '', $value->name);
             $tmpparam = optional_param($tmpparam, '', PARAM_RAW);
-            if ($value->type == "Text Area") {
+            if ($value->type == get_string('textarea', 'local_pages')) {
```

Comment:
As long as $value->type is always going to match whatever is in the language string textarea. What happens if someone changes the language string, does the value in $value->type change? If not, perhaps a bunch of string constants?

If case could be an issue, $value->type could be converted to lower or upper and then evaluated against an all caps or lowercase string i.e. 

```
// Or the classy way const SOME_CONSTANT = 'somevalue';
define('SOME_CONSTANT', 'somevalue');
$thetype = core_text::strtolower($value->type);
if ($thetype == SOME_CONSTANT) {

}

Reply:
I think it's only used for the dropdowns when setting up the form and then mainly in the background after that. But yeah, I wondered about that too. Challenge I guess is/was to support translations too. Looks like the plugin has been translated into a few languages already where it allows for it: [Plugin Translations](https://moodle.org/plugins/translations.php?plugin=local_pages)

May just leave for now, but open for review - perhaps @digitalskill can advise/fix here too

Reply:
Looks fine to me.
Field form names in html are the same regardless of language (textarea is a HTML term - so no language file needed - "text area" (its already a HTML constant) but can see your side as well, variables are always usefull for code readability.

I'll add this to the 39_stable branch.

Reply:
Thanks man - will merge it into 35_STABLE